pkg/net/tlsconfig/server.go

```diff
@@ -59,9 +59,22 @@ func (constructor Constructor) Annotate() fx.Option {
 			constructor.provideTLSConfig,
 			fx.ResultTags(name),
 		),
+		fx.Annotate(
+			constructor.provideServerTLSConfig,
+			fx.ResultTags(name),
+		),
 	)
 }
 
+func (constructor Constructor) provideServerTLSConfig(unmarshaller config.Unmarshaller) (ServerTLSConfig, error) {
+	config := constructor.DefaultConfig
+	if err := unmarshaller.UnmarshalKey(constructor.ConfigKey, &config); err != nil {
```

Comment:
Why repeat this code? Can't we just provide this to provide TLS function and remove repetition? Or have the provide TLS function return config as well.